src/main/java/com/hazelcast/jdbc/HazelcastConfigFactory.java

```diff
@@ -187,6 +191,19 @@ private static void gcpConfig(ClientConfig clientConfig, String property, String
                 .setProperty(property, value);
     }
 
+    private static ClientSqlResubmissionMode parseSqlResubmissionMode(JdbcUrl url, String key, ClientSqlResubmissionMode defaultVal) {
+        String value = url.getProperty(key);
+        if (value == null) {
+            return defaultVal;
+        }
+        try {
+            return ClientSqlResubmissionMode.valueOf(value);
+        } catch (IllegalArgumentException e) {
+            String message = String.format("'%s' not valid for SQL resubmission mode, '%s'", key, value);
```

Comment:
key/value is swapped in the msg:
```suggestion
            String message = String.format("'%s' not a valid value for '%s'", value, key);
```

We should also cover it in a test where it would be more obvious.

Reply:
Done